### ISSUE TITLE
[MDC] Block Components with PascalCase are not working

### ISSUE THREAD

**Issue by aaharu** [nuxt/content#2497]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v18.18.0
- Nuxt Version:     3.9.1
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  npm@9.4.2
- Builder:          -
- User Config:      devtools, modules
- Runtime Modules:  @nuxt/content@2.10.0
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-zvrsnm?file=content%2Fblock.md

### Describe the bug

MDC's block components failed to resolve components when I wrote it in PascalCase.
Is this by design?

`components/content/HeroHero.vue`

`::HeroHero`\(Block components\) doesn't work, but `:HeroHero`\(Inline components\) works fine.

```md
::hero-hero
It works!

#description
hero-hero
::

::HeroHero
☑️ `<HeroHero>:[Vue warn]: Failed to resolve component: Herohero`

#description
HeroHero
::
```

**Comment by farnabaz** [comment]
Thanks for the reproduction. It is fixed in https://github.com/nuxtlabs/remark-mdc/commit/9d2d00e55ee17d3f8f594bc8b457508f95ce1424 and will be released in next patch.

**Comment by aaharu** [comment]
This issue has been resolved by [remark-mdc v3.0.1](https://www.npmjs.com/package/remark-mdc/v/3.0.1).
https://github.com/nuxtlabs/remark-mdc/releases/tag/v3.0.1

Thanks!
